app/src/main/kotlin/com/mapbox/vision/teaser/MainActivity.kt

```diff
@@ -1,55 +1,531 @@
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import androidx.core.text.HtmlCompat
+import androidx.core.view.setPadding
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants
+import com.mapbox.services.android.navigation.v5.utils.DistanceFormatter
+import com.mapbox.services.android.navigation.v5.utils.LocaleUtils
 import com.mapbox.vision.VisionManager
 import com.mapbox.vision.VisionReplayManager
-import com.mapbox.vision.mobile.core.models.FrameStatistics
+import com.mapbox.vision.mobile.core.interfaces.VisionEventsListener
+import com.mapbox.vision.mobile.core.models.Country
+import com.mapbox.vision.mobile.core.models.classification.FrameSignClassifications
+import com.mapbox.vision.mobile.core.models.detection.DetectionClass
+import com.mapbox.vision.mobile.core.models.position.VehicleState
+import com.mapbox.vision.mobile.core.models.road.LaneDirection
+import com.mapbox.vision.mobile.core.models.road.LaneEdgeType
+import com.mapbox.vision.mobile.core.models.road.RoadDescription
+import com.mapbox.vision.mobile.core.utils.SystemInfoUtils
+import com.mapbox.vision.performance.ModelPerformance
+import com.mapbox.vision.performance.ModelPerformanceMode
+import com.mapbox.vision.performance.ModelPerformanceRate
+import com.mapbox.vision.teaser.view.show
 import com.mapbox.vision.safety.VisionSafetyManager
 import com.mapbox.vision.teaser.MainActivity.VisionManagerMode.Camera
 import com.mapbox.vision.teaser.MainActivity.VisionManagerMode.Replay
+import com.mapbox.vision.safety.core.VisionSafetyListener
+import com.mapbox.vision.safety.core.models.CollisionDangerLevel
+import com.mapbox.vision.safety.core.models.CollisionObject
+import com.mapbox.vision.safety.core.models.RoadRestrictions
 import com.mapbox.vision.teaser.ar.ArMapActivity
+import com.mapbox.vision.teaser.models.UiSign
 import com.mapbox.vision.teaser.recorder.RecorderFragment
 import com.mapbox.vision.teaser.replayer.ArReplayNavigationActivity
 import com.mapbox.vision.teaser.replayer.ReplayModeFragment
-import com.mapbox.vision.teaser.view.BaseTeaserActivity
+import com.mapbox.vision.teaser.utils.SoundsPlayer
+import com.mapbox.vision.teaser.utils.classification.SignResources
+import com.mapbox.vision.teaser.utils.classification.Tracker
+import com.mapbox.vision.teaser.utils.dpToPx
 import com.mapbox.vision.teaser.view.hide
-import com.mapbox.vision.teaser.view.show
+import com.mapbox.vision.utils.VisionLogger
 import com.mapbox.vision.view.VisionView
 import com.mapbox.vision.view.VisualizationMode
 import kotlinx.android.synthetic.main.activity_main.*
 
-class MainActivity : BaseTeaserActivity(), ReplayModeFragment.OnSelectModeItemListener {
+class MainActivity : AppCompatActivity(), ReplayModeFragment.OnSelectModeItemListener {
 
     enum class VisionManagerMode {
         Camera,
         Replay,
     }
 
-    private var mode = Camera
+    enum class VisionMode {
```

Comment:
And this class selects mode for the view, lets think about better name for the enum to indicate that. 

@mapbox/vision-android  WDYT `ViewMode`/`Feature`/`VisionFeature`/`ActiveFeature`

Reply:
I like VisionFeature

Reply:
fixed to VisionFeature for now